annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -409,39 +413,50 @@ private String resolveHgvspShort() {
-                            cPos = Integer.parseInt(m.group(1));
-                            cPos = cPos < 1 ? 1 : cPos;
-                            pPos = (cPos + cPos % 3) / 3;
-                            hgvsp = "p.X" + String.valueOf(pPos) + "_splice";
-                            break;
-                        }
-                    }
+            else if (canonicalTranscript.getHgvsc() != null && spliceSiteVariants.contains(canonicalTranscript.getConsequenceTerms().get(0))) {
+                Integer cPos = 0;
+                Integer pPos = 0;
+                Matcher m = cDnaExtractor.matcher(canonicalTranscript.getHgvsc());
+                if (m.matches()) {
+                    cPos = Integer.parseInt(m.group(1));
+                    cPos = cPos < 1 ? 1 : cPos;
+                    pPos = (cPos + cPos % 3) / 3;
+                    hgvsp = "p.X" + String.valueOf(pPos) + "_splice";
                 }
             }
             else {
                 // try to salvage using protein_start, amino_acids, and consequence_terms
-                try {
-                    String[] aaParts = canonicalTranscript.getAminoAcids().split("/");
-                    hgvsp = aaParts[0] + canonicalTranscript.getProteinStart();
-                    if (canonicalTranscript.getConsequenceTerms() != null && canonicalTranscript.getConsequenceTerms().get(0).equals("frameshift_variant")) {
-                        hgvsp += "fs";
-                    }
-                    else {
-                        hgvsp += aaParts[1];
-                    }
+                hgvsp = resolveHgvspShortFromAAs();
+            }
+        }
+        return hgvsp;
+    }
+    
+    private String resolveHgvspShortFromAAs() {
+        String hgvsp = "";
+        try {
+            String[] aaParts = canonicalTranscript.getAminoAcids().split("/");
+            if (canonicalTranscript.getConsequenceTerms() != null && canonicalTranscript.getConsequenceTerms().get(0).equals("inframe_insertion")) {
+                hgvsp = aaParts[1].substring(0,1) + canonicalTranscript.getProteinStart() + "_" + aaParts[1].substring(1, 2) + "ins" + 
+                        canonicalTranscript.getProteinEnd() + aaParts[1].substring(2);
+            }
+            else if (canonicalTranscript.getConsequenceTerms() != null && canonicalTranscript.getConsequenceTerms().get(0).equals("inframe_deletion")) {
+                hgvsp = aaParts[0] + "del";
+            }
+            else {
+                hgvsp = aaParts[0] + canonicalTranscript.getProteinStart();
+                if (canonicalTranscript.getConsequenceTerms() != null && canonicalTranscript.getConsequenceTerms().get(0).equals("frameshift_variant")) {
+                    hgvsp += "fs";
                 }
-                catch (NullPointerException e) {
-                    log.debug("Failed to salvage HGVSp_Short from protein start, amino acids, and consequence terms");
+                else {
+                    hgvsp += aaParts[1];
```

Comment:
Is this proper behavior? I think better to just accept that it didn't annotate properly and keep the log? Maybe upgrade the log from debug level to error level.